File: LeetCode/434. Number of Segments in a String/solution.py
class Solution(object):
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        return len(s.split())

    def countSegments1(self, s):
        preSpace, count = True, 0
        for c in s:
            if not preSpace and c == ' ':
                preSpace = True
            elif preSpace and c != ' ':
                preSpace = False
                count += 1
        return count

    def countSegments2(self, s):
        count = 0
        for i in range(len(s)):
            if s[i] != ' ' and (i == 0 or s[i - 1] == ' '):
                count += 1
        return count

File: LeetCode/434. Number of Segments in a String/Solution.java
public class Solution {
    public int countSegments(String s) {
        boolean preSpace = true;
        int count = 0;
        for (char c : s.toCharArray()) {
            if (!preSpace && c == ' ') {
                preSpace = true;
            } else if (preSpace && c != ' ') {
                preSpace = false;
                count++;
            }
        }
        return count;
    }


    public static void main(String[] args) {
        String s = ", , , ,        a, eaefa";
        System.out.println(new Solution().countSegments(s));
        System.out.println((s.split(" +").length));
    }
}